src/lib.rs

```diff
@@ -1,314 +1,40 @@
-//! Findex is a cryptographic algorithm allowing to securely maintain an encrypted index.
-//!
-//! It uses a generic Dictionary Encryption Scheme (Dx-Enc) as building block to implement a
-//! Multi-Map Encryption Scheme (MM-Enc). A Graph Encryption Scheme (Gx-Enc) is then built on top
-//! of the MM-Enc scheme and finally an `Index` trait built on top of this Gx-Enc scheme allows
-//! indexing both `Data` and `Keyword`s.
-//!
-//! The `Index` traits is not a cryptographic one. It is used to simplify the interface and to hide
-//! the details of the cryptographic implementation when it is possible.
+#![allow(clippy::type_complexity)]
```

Comment:
pedantic is way too much, some lints I disagree with, same for unreachable_pub: to me the number of possible combinations with pub modules, pub crates modules, pub reexports, pub declarations inside a module, pub crate declarations inside a module is waaay too great. So I try keeping myself to a simpler rule:
- all useful declarations inside a module are `pub`
- each module is declared private and useful stuffs are re-exported publicly by hand.

There is no possible combination anymore, and exports are explicit.

Restricting `unwrap` raise warning in tests so there too I disagree: unwrapping is way better than raising an error in tests since it returns the line at which the error occurred.

Reply:
I'm in favor of denying all functions that can panic in production code, except within tests. Since you're the primary developer on the repository, choose obviously the lints that works for you. However, in a collaborative context, I would definitely choose enforcing at least the `unwrap_used` lint. It's almost free. But thanks to have changed the actual clippy behavior.